pkg/daemon/update.go

```diff
@@ -797,9 +797,11 @@ func reconcilable(oldConfig, newConfig *mcfgv1.MachineConfig) (*machineConfigDif
 
 	// FIPS section
 	// We do not allow update to FIPS for a running cluster, so any changes here will be an error
-	if err := checkFIPS(oldConfig, newConfig); err != nil {
+	if err := checkFIPS(newConfig.Spec.FIPS); err != nil {
 		return nil, err
 	}
+	// if the oldConfig somehow drifted from the state on the node, we want to ignore it when we calculate newMachineConfigDiff
+	oldConfig.Spec.FIPS = newConfig.Spec.FIPS
```

Comment:
I think it might be better to drop this and remove checking FIPS from `newMachineConfigDiff`, but that change would affect more codepaths (specifically `RunFirstbootCompleteMachineconfig`), so I made the less invasive change

Reply:
Hmm.  Ah, yes.  OK I think what's happening here is on firstboot we should detect that the machine is already in FIPS state when we synthesize the current state.

(And this would all be cleaned up and go away if we canonically represented "fips" by the `fips=1` kernel argument as proposed in https://github.com/coreos/ignition/issues/1323 )